text/0000-alternatives-in-patterns.md

```diff
@@ -0,0 +1,78 @@
+- Feature Name: Alternatives in patterns
+- Start Date: 2016-02-14
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+Extend the pattern syntax for alternatives in `match` statement, allow alternatives for pattern matching in `let` and `if let` statements.
+
+# Motivation
+[motivation]: #motivation
+
+Rust allows alternatives ( `|` ) in pattern syntax for `match`, but only for 'top-level' of pattern.
+This aims to reduce verbosity in certain examples and increase expressiveness.
+
+Also, this RFC proposes to allow alternatives in `let` or `if let` statements.
+
+# Detailed design
+[design]: #detailed-design
+
+## Expand usage of alternatives to 'deeper levels'
+Imagine a type:
+```rust
+#[derive(Clone, Copy)] struct NewType (Result<String,String>, u8);
+```
+
+Exhaustive `match` statement for this type would look like this:
+```rust
+match new_type {
+    NewType(Ok(e), num) | NewType(Err(e), num) => println!("ok with {}: {}", num, e)
+}
+```
+
+This RFC proposes a following replacement:
+```rust
+match new_type {
+    NewType(Ok(e) | Err(e), num) => println!("ok with {}: {}", num, e)
+}
+```
```

Comment:
No, it can't. Because `e` is used in arm. If you don't want to use, then it may be written as `NewType(_, num)`